rubix-tests/src/test/java/com/qubole/rubix/tests/TestThriftServerJVM.java

```diff
@@ -179,4 +179,30 @@ public void testCreateBookKeeperClient_OnNotRunningPort() throws IOException, In
     String host = "localhost";
     bookKeeperFactory.createBookKeeperClient(host, conf);
   }
+
+  @Test(enabled = true)
+  public void testBookKeeperHealth() throws IOException, InterruptedException, TTransportException, TException
+  {
+    String host = "localhost";
+
+    RetryingBookkeeperClient client;
+    client = bookKeeperFactory.createBookKeeperClient(host, conf);
+
+    boolean isBookKeeperAlive = client.isBookKeeperWorking();
+    assertTrue(isBookKeeperAlive == true, "BookKeeper is not Alive");
+  }
+
+  @Test(enabled = true, expectedExceptions = org.apache.thrift.shaded.transport.TTransportException.class)
+  public void testBookKeeperHealthAtWrongPort() throws IOException, InterruptedException, TTransportException, TException
+  {
+    String host = "localhost";
+
+    //1234 is a random port to test if BookKeeperHealth is detecting that bookkeeper is down or not
+    CacheConfig.setServerPort(conf, 1234);
+    RetryingBookkeeperClient client;
+    client = bookKeeperFactory.createBookKeeperClient(host, conf);
+
+    boolean isBookKeeperAlive = client.isBookKeeperWorking();
+    assertTrue(isBookKeeperAlive == false, "Showing BookKeeper Alive, It shouldn't be up at this port");
```

Comment:
Remove this assert. This will never get executed as the previous line is going to throw an exception